src/python/pants/init/specs_calculator.py

```diff
@@ -50,13 +50,13 @@ def calculate_specs(
     if not changed_options.provided:
         return specs
 
-    git = get_git()
-    if not git:
+    git_result = get_git()
```

Comment:
As mentioned in my other comment: if this `@rule` gains any dependencies (due to using the `system_binaries` infrastructure), you'll need to actually invoke it as a `@rule`, which will look like:
```python
git_result = session.product_request(GitResult, [])
```
...and then delete the `GitInitialized` stuff.